metaflow/flowspec.py

```diff
@@ -529,7 +530,7 @@ def _is_primitive_type(item):
             )
 
         value = item if _is_primitive_type(item) else reprlib.Repr().repr(item)
-        return basestring(value)
+        return basestring(value)[:MAXIMUM_VALUE_CHARACTERS]
```

Comment:
Repr already has a default limit of 30 already - I imagine this change only affects primitives right now. I would recommend setting the value of `maxstring` to `MAX_CHARS` so that if the value of `MAX_CHARS` is modified in the future - you can expect a consistent response if `repr` is in action.

Reply:
Repr doc is ambiguous, the `maxstring=30` for Repr means that the string inside the represented result is <=30 (it doesn't truncate the entire representation to maxstring.  for example, 

```
reprlib.Repr().repr({"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa": "bbbbbbbbbbbbbbbbbbbbbbb"})
"{'aaaaaaaaaaaa...aaaaaaaaaaaaa': 'bbbbbbbbbbbbbbbbbbbbbbb'}"
```